--- FUGUITA.MP.orig	Tue Dec  3 23:18:57 2013
+++ FUGUITA.MP	Tue Dec  3 23:25:30 2013
@@ -25,7 +25,7 @@
 option		NTFS		# NTFS support
 option		HIBERNATE	# Hibernate support
 
-config		bsd	swap generic
+config		bsd	root on rd0a swap on rd0b and wd0b and sd0b
 
 mainbus0 at root
 
@@ -33,7 +33,7 @@
 cpu0	at mainbus?
 ioapic*	at mainbus?
 isa0	at mainbus0
-isa0	at pcib?
+#isa0	at pcib?
 isa0	at amdpcib?
 isa0	at tcpcib?
 pci*	at mainbus0
@@ -281,7 +281,7 @@
 #btsco*	at bthub? 		# Bluetooth audio support
 #audio*	at btsco?
 
-isadma0	at isa?
+#isadma0	at isa?
 
 option	WSDISPLAY_COMPAT_USL		# VT handling
 option	WSDISPLAY_COMPAT_RAWKBD		# provide raw scancodes; needed for X11
@@ -319,7 +319,7 @@
 com2	at isa? port 0x3e8 irq 5
 com3	at isa? disable port 0x2e8 irq 9 # (conflicts with some video cards)
 
-com*	at pcmcia?			# PCMCIA modems/serial ports
+#com*	at pcmcia?			# PCMCIA modems/serial ports
 com*	at puc?
 
 # options CY_HW_RTS
@@ -354,7 +354,7 @@
 iha*	at pci?				# Initio Ultra/UltraWide SCSI controllers
 ips*	at pci?				# IBM ServeRAID controllers
 isp*	at pci?				# Qlogic ISP [12]0x0 SCSI/FibreChannel
-aic*	at pcmcia?			# PCMCIA Adaptec 152[02] SCSI
+#aic*	at pcmcia?			# PCMCIA Adaptec 152[02] SCSI
 #esp*	at pcmcia?			# PCMCIA based NCR 53C9X SCSI
 siop*	at pci?				# NCR 538XX SCSI controllers
 #adv*	at pci?				# AdvanSys 1200A/B and ULTRA SCSI
@@ -372,16 +372,16 @@
 ses*	at scsibus?			# SCSI enclosure services
 uk*	at scsibus?			# unknown SCSI
 
-fdc0	at isa? port 0x3f0 irq 6 drq 2	# standard PC floppy controllers
+#fdc0	at isa? port 0x3f0 irq 6 drq 2	# standard PC floppy controllers
 #fdc1	at isa? port 0x370
-fd*	at fdc? flags 0x00
+#fd*	at fdc? flags 0x00
 
 # IDE controllers
 pciide* at pci? flags 0x0000
 
 wdc0	at isa? disable port 0x1f0 irq 14 flags 0x00
 wdc1	at isa? disable port 0x170 irq 15 flags 0x00
-wdc*	at pcmcia?
+#wdc*	at pcmcia?
 
 # IDE hard drives
 wd*	at wdc? flags 0x0000
@@ -399,10 +399,10 @@
 fxp*	at cardbus?			# Intel PRO/100 ethernet
 ne*	at pci?				# NE2000-compat ethernet
 ep*	at pci?				# 3C59x ethernet
-ne*	at pcmcia?			# PCMCIA based NE2000 ethernet
-ep*	at pcmcia?			# PCMCIA based 3C5xx ethernet
-sm*	at pcmcia?			# PCMCIA based sm ethernet
-xe*	at pcmcia?			# Xircom ethernet
+#ne*	at pcmcia?			# PCMCIA based NE2000 ethernet
+#ep*	at pcmcia?			# PCMCIA based 3C5xx ethernet
+#sm*	at pcmcia?			# PCMCIA based sm ethernet
+#xe*	at pcmcia?			# Xircom ethernet
 xl*	at pci?				# 3C9xx ethernet
 xl*	at cardbus?			# 3C575/3C656 ethernet
 rl*	at pci?				# RealTek 81[23]9 ethernet
@@ -468,10 +468,10 @@
 bwi*	at pci?				# Broadcom AirForce (802.11b/g)
 bwi*	at cardbus?			# Broadcom AirForce (802.11b/g)
 wi*	at pci?				# WaveLAN IEEE 802.11DS
-wi*	at pcmcia?			# WaveLAN IEEE 802.11DS
+#wi*	at pcmcia?			# WaveLAN IEEE 802.11DS
 an*	at pci?				# Aironet IEEE 802.11DS
-an*	at pcmcia?			# Aironet IEEE 802.11DS
-cnw*	at pcmcia?			# Xircom Netwave
+#an*	at pcmcia?			# Aironet IEEE 802.11DS
+#cnw*	at pcmcia?			# Xircom Netwave
 iwi*	at pci?				# Intel PRO/Wireless 2200BG/2915ABG
 wpi*	at pci?				# Intel PRO/Wireless 3945ABG
 iwn*	at pci?				# Intel WiFi Link 4965/5000/1000/6000
@@ -483,7 +483,7 @@
 pgt*	at cardbus?			# Prism54 (only full-mac varients)
 malo*	at pci?				# Marvell Libertas
 malo*	at cardbus?			# Marvell Libertas
-malo*	at pcmcia?			# Marvell 88W8385
+#malo*	at pcmcia?			# Marvell 88W8385
 
 # Media Independent Interface (mii) drivers
 exphy*	at mii?				# 3Com internal PHYs
@@ -601,6 +601,16 @@
 # mouse & keyboard multiplexor pseudo-devices
 pseudo-device	wsmux		2
 pseudo-device	crypto		1
+
+# For FuguIta
+option		RAMDISK_HOOKS
+option		MINIROOTSIZE=4480
+option		PPPOE_TERM_UNKNOWN_SESSIONS
+option		MULTIPROCESSOR
+pseudo-device	rd	1	# ramdisk
+pseudo-device	vnd	6	# vnode disk devices
+
+cpu*		at mainbus?
 
 # Virtio devices
 virtio*		at pci?		# Virtio PCI device
